flow/record/adapter/sqlite.py

```diff
@@ -0,0 +1,234 @@
+    for column_name, fieldset in descriptor.get_all_fields().items():
+        if column_name in column_names:
+            continue
+        column_type = FIELD_MAP.get(fieldset.typename, "TEXT")
+        column_defs.append(f"  ALTER TABLE [{table_name}] ADD COLUMN [{column_name}] {column_type}")
+
+    # No missing columns
+    if not column_defs:
+        return None
+
+    # Add the new columns
+    sql = ";\n".join(column_defs)
+    con.executescript(sql)
+
+
+def db_insert_record(con: sqlite3.Connection, record: Record) -> None:
+    """Insert a record into the database."""
+    table_name = record._desc.name
+    rdict = record._asdict()
+
+    # Construct placeholder SQL query
+    column_names = ", ".join(f'"{name}"' for name in rdict.keys())
+    value_placeholder = ", ".join(["?"] * len(rdict))
+    sql = f"""INSERT INTO "{table_name}" ({column_names}) VALUES ({value_placeholder})"""
+
+    # Convert values to str() for types we don't support
+    values = []
+    for value in rdict.values():
+        if isinstance(value, datetime):
+            value = value.isoformat()
+        elif not (isinstance(value, (bytes, int, bool, float)) or value is None):
+            value = str(value)
+        values.append(value)
+
+    # Insert record into database
+    logger.debug(sql)
+    logger.debug(values)
+    con.execute(sql, values)
+
+
+class SqliteReader(AbstractReader):
+    def __init__(self, path, **kwargs):
+        self.descriptors_seen = set()
+        self.con = sqlite3.connect(path)
+        self.count = 0
+        self.batch_size = 1000
+        self.rewrite = {}
+
+    def table_names(self) -> list[str]:
+        """Return a list of table names in the database."""
+        records = self.con.execute("SELECT name FROM sqlite_master WHERE type='table'").fetchall()
+        return [record[0] for record in records]
+
+    def read_table(self, table_name: str) -> Iterator[Record]:
+        """Read a table from the database and yield records."""
+
+        # flow.record is quite strict with what is allowed in fieldnames or decriptor name.
+        # While SQLite is less strict, we need to sanitize the names to make them compatible.
+        table_name_org = table_name
+        table_name = sanitized_name(table_name)
+
+        schema = self.con.execute(
+            "SELECT c.type, c.name FROM pragma_table_info(?) c",
+            [table_name_org],
+        ).fetchall()
+
+        fields = []
+        fnames = []
+        fname_to_type = {}
+        for idx, row in enumerate(schema):
+            ftype, fname = row
+            fname = sanitized_name(fname)
+            ftype = SQLITE_FIELD_MAP.get(ftype, "string")
+            fname_to_type[fname] = ftype
+            if fname not in RESERVED_FIELDS:
+                fields.append((ftype, fname))
+            fnames.append(fname)
+
+        descriptor_cls = RecordDescriptor(table_name, fields)
+        cursor = self.con.execute(f'SELECT * FROM "{table_name_org}"')
```

Comment:
Same thing here about the quotes, except that you now chose to use `'` for the Python string.

Reply:
backticked